src/main/java/com/workshop/demo/controllers/RestaurantController.java

```diff
@@ -42,8 +43,7 @@ public Integer getScore(
         return score;
     }
 
-    @PostMapping
-    @PreAuthorize("hasRole('USER')")
+    @PostMapping("/add")
     public Restaurant addRestaurant(@Valid @RequestBody RestaurantRequest restaurantRequest,
```

Comment:
Currently the return type is Restaurant, which is the restaurant that was created. However, it cannot handle error cases (e.g name already taken) and does not give user any error messages when it fails. Please consider changing this to something like RestaurantResponse.
```
class RestaurantResponse {
  Restaurant restaurant; // populated if success
  boolean success; 
  string message; // can be used to store error message.
  HttpStatus httpStatus;
}
```
If the addRestaurant is successful, set restaurant object, success true and 200 httpStatus. Otherwise set the corresponding error code, error message.